tests/providers/docker/operators/test_docker.py

```diff
@@ -32,23 +32,32 @@
 
 
 class TestDockerOperator(unittest.TestCase):
-    @mock.patch('airflow.providers.docker.operators.docker.TemporaryDirectory')
-    @mock.patch('airflow.providers.docker.operators.docker.APIClient')
-    def test_execute(self, client_class_mock, tempdir_mock):
-        host_config = mock.Mock()
-        tempdir_mock.return_value.__enter__.return_value = '/mkdtemp'
-
-        client_mock = mock.Mock(spec=APIClient)
-        client_mock.create_container.return_value = {'Id': 'some_id'}
-        client_mock.create_host_config.return_value = host_config
-        client_mock.images.return_value = []
-        client_mock.attach.return_value = ['container log']
-        client_mock.logs.return_value = ['container log']
-        client_mock.pull.return_value = {"status": "pull log"}
-        client_mock.wait.return_value = {"StatusCode": 0}
+    def setUp(self):
+        self.tempdir_patch = mock.patch(
+            'airflow.providers.docker.operators.docker.TemporaryDirectory',
+        )
```

Comment:
```suggestion
        self.tempdir_patcher = mock.patch(
            'airflow.providers.docker.operators.docker.TemporaryDirectory',
        ).patcher()
```

Reply:
See: https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop

Reply:
Thank you, sir, worked perfectly! There will never come a day when I fully read the mock documentation, always find smth new there :D